packages/bpk-component-chip/src/BpkIconChip.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+The icon chip component is just a selectable chip that's
+been hard coded to have no text or trailing icon and padding/margin
+to match.
+*/
+
+import type { ReactNode } from 'react';
+
+import { cssModules } from '../../bpk-react-utils';
+
+import BpkSelectableChip from './BpkSelectableChip';
+import type { CommonProps as Props } from './commonTypes';
+import STYLES from './BpkSelectableChip.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+export type BpkIconChipProps = {
+  leadingAccessoryView: ReactNode;
+} & Omit<Props, 'trailingAccessoryView' | 'leadingAccessoryView' | 'children'>;
+
+const BpkIconChip = ({
+ className,
+ ...rest
+}: BpkIconChipProps) => {
+  const classNames = getClassName(`bpk-chip--icon-only`, className);
```

Comment:
This is going to be interesting one and will need some thinking, as we ideally don't want to introduce a className prop that we then send to override another Backpack component as it introduces a non-determinisim risk by doing so

Reply:
Having a bit of think about this, I am wondering if we actually need a separate component here and instead to the existing `BpkSelectableChip` we just simply add a prop `iconOnly` similar to the `BpkButton` which will just apply the icon only CSS

Reply:
I've updated to use `!children` implies icon-only in BpkSelectableChip. This avoids className issues but might make it harder to extend BpkSelectableChip in future, what do you think? `iconOnly` may be a better long term solution?

I see the other chip types just hardcode certain props so it seemed consistent to have a new component. Happy to remove though if that's preferred.